kazoo/protocol/connection.py

```diff
@@ -614,7 +614,9 @@ def _connect_attempt(self, host, hostip, port, retry):
 
         try:
             self._xid = 0
-            read_timeout, connect_timeout = self._connect(host, hostip, port)
+            read_timeout, connect_timeout = self._connect(
+                host, hostip, port, timeout=retry.cur_delay
```

Comment:
The logic looks good, but the default initial `_cur_delay` is 100ms.  Do we really want to keep it that low?